### PR TITLE
Api backend

### PR DESCRIPTION
**Pull request by dakota002**

### CODE REVIEW

**Comment by dakota002** [comment]
This update will include adding some validation to the subject line input, which applies to the UI and API

Invalid:
image

Valid:
image

File: app/routes/circulars/circulars.server.ts [hunk 1]
```diff
@@ -63,8 +63,10 @@ export async function list({
 
   // Calculate pagination assuming that last === the number of records.
   const totalPages = Math.ceil(last / limit)
-  const firstIdOnPage = Math.max(last - page * limit + 1, 1)
-
+  const firstIdOnPage = Math.max(
```
Comment (dakota002) [inline app/routes/circulars/circulars.server.ts:66]: This change was mainly for an issue locally where the `limit` far exceeds the `last`, and was causing an issue where no circulars would show up

File: app/routes/circulars/index.mdx [hunk 2]
```diff
@@ -41,36 +42,11 @@ Please follow the [style guidelines](/docs/styleguide). In particular, the email
...
-  <li className="grid-col-2">SDSS</li>
-  <li className="grid-col-2">SGR</li>
-  <li className="grid-col-2">Swift</li>
-  <li className="grid-col-2">SWIFT</li>
-  <li className="grid-col-2">Virgo</li>
-  <li className="grid-col-2">VLA</li>
-  <li className="grid-col-2">VLBI</li>
-  <li className="grid-col-2">XRB</li>
-  <li className="grid-col-2">XTR</li>
+  {validSubjectKeywords?.map((keyword) => (
```
Comment (dakota002) [inline app/routes/circulars/index.mdx:45]: any new keywords can just be added to the `validSubjectKeywords` list and this page will automatically update

File: sandbox-seed.json [hunk 3]
```diff
@@ -1,109 +1,130 @@
 {
   "circulars": [
     {
+      "dummy": 0,
```
Comment (lpsinger) [inline sandbox-seed.json:4]: I fixed this on the main branch. Please rebase.

File: app/routes/circulars/circulars.server.ts (hunk 1 above)
Comment (lpsinger) [reply app/routes/circulars/circulars.server.ts:66]: OK, would you please split that out as a separate PR?

File: app/routes/circulars/index.mdx (hunk 2 above)
Comment (lpsinger) [reply app/routes/circulars/index.mdx:45]: Please split the keyword validation out to a separate PR (separate from adding the API endpoints).

File: app/routes/circulars/circulars.server.ts (hunk 1 above)
Comment (dakota002) [reply app/routes/circulars/circulars.server.ts]: I think you actually solved this in a previous commit, my branch was just out of date

File: sandbox-seed.json (hunk 3 above)
Comment (dakota002) [reply sandbox-seed.json]: Sweet, rebased

File: app/routes/circulars/index.mdx (hunk 2 above)
Comment (dakota002) [reply app/routes/circulars/index.mdx:45]: Okay, I originally have that in the api/circulars file, should I keep the validation in that file, or move them into another file like utils?

Comment (dakota002) [reply app/routes/circulars/index.mdx:45]: Broke it out here https://github.com/nasa-gcn/gcn.nasa.gov/pull/523

**Review by lpsinger — CHANGES_REQUESTED** [review]
I don't think that we are ready to add the API yet. We first need to figure out how users are going to authenticate for API access.

**Pull request closed without merging**